src/chainstate/burn/operations/mod.rs

```diff
@@ -447,3 +472,398 @@ pub fn parse_u32_from_be(bytes: &[u8]) -> Option<u32> {
 pub fn parse_u16_from_be(bytes: &[u8]) -> Option<u16> {
     bytes.try_into().ok().map(u16::from_be_bytes)
 }
+
+// serialization functions
```

Comment:
Just a thought -- it might make sense to put these methods into a file somewhere in `stacks-common` so we can use them in other contexts.  Maybe `stacks-common/src/codec/serde.rs` or something?